src/engine/controls/loopingcontrol.cpp

```diff
@@ -512,6 +530,59 @@ double LoopingControl::getSyncPositionInsideLoop(double dRequestedPlaypos, doubl
     return dSyncedPlayPos;
 }
 
+void LoopingControl::setSavedLoop(CuePointer pCue, bool toggle) {
+    qDebug() << "setSavedLoop" << toggle;
+
+    if(!pCue) {
+        if(m_pCue) {
+            m_pCue->deactivate();
+            m_pCue.reset();
+        }
+        clearActiveBeatLoop();
+        return;
+    }
+    DEBUG_ASSERT(pCue->getType() == Cue::LOOP);
+
+    double startPosition = pCue->getPosition();;
+    double endPosition = startPosition + pCue->getLength();
+    if(startPosition < 0 || startPosition >= endPosition) {
+        if(m_pCue) {
+            m_pCue->deactivate();
+            m_pCue.reset();
+        }
+        clearActiveBeatLoop();
+        return;
+    }
+
+    LoopSamples loopSamples = m_loopSamples.getValue();
+    bool loopSamplesChanged = loopSamples.start != startPosition || loopSamples.end != endPosition;
+    if (m_pCue != pCue || loopSamplesChanged) {
+        if(m_pCue != pCue) {
+            if(m_pCue) {
+                m_pCue->deactivate();
+            }
+            m_pCue = pCue;
+        }
+        if(loopSamplesChanged) {
+            // Copy saved loop parameters to active loop
+            loopSamples.start = startPosition;
+            loopSamples.end = endPosition;
+            loopSamples.seekMode = LoopSeekMode::None;
+            clearActiveBeatLoop();
+            m_loopSamples.setValue(loopSamples);
+            m_pCOLoopStartPosition->set(loopSamples.start);
+            m_pCOLoopEndPosition->set(loopSamples.end);
+            setLoopingEnabled(true);
+        } else if(m_pCue && m_bLoopingEnabled) {
+            // Currently active loop became a saved loop, mark it as active
+            m_pCue->activate();
```

Comment:
Why is this in the else branch?

Reply:
This happens when we already have an active loop and the user presses a hotcue button. In that case, the loop will stay unchanged, but since it's a saved loop now, we want to mark it as active so that we can give visual feedback on screen/controller that a saved loop is now active.